os/kernel/init/os_start.c

```diff
@@ -384,8 +389,15 @@ void os_start(void)
 		 */
 
 		up_allocate_heap(&heap_start, &heap_size);
+#ifdef CONFIG_BOOTMEM
```

Comment:
Does this logic of bootmem applies only to CONFIG_MM_KERNEL_USRHEAP_INIT ?  I believe it holds good for other configurations as well. Can you please cross check ?